packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts

```diff
@@ -0,0 +1,148 @@
+import { caip2 } from '@shapeshiftoss/caip'
+import {
+  bip32ToAddressNList,
+  CosmosSignTx,
+  CosmosTx,
+  CosmosWallet
+} from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+import BigNumber from 'bignumber.js'
+
+import { ChainAdapter as IChainAdapter } from '../../api'
+import { ErrorHandler } from '../../error/ErrorHandler'
+import { toPath } from '../../utils'
+import { ChainAdapterArgs, CosmosSdkBaseAdapter } from '../CosmosSdkBaseAdapter'
+
+export class ChainAdapter
+  extends CosmosSdkBaseAdapter<ChainTypes.Cosmos>
+  implements IChainAdapter<ChainTypes.Cosmos>
+{
+  protected readonly supportedChainIds = ['cosmos:cosmoshub-4', 'cosmos:vega-testnet']
+  protected readonly chainId = this.supportedChainIds[0]
+  public static readonly defaultBIP44Params: BIP44Params = {
+    purpose: 44,
+    coinType: 118,
+    accountNumber: 0
+  }
+
+  constructor(args: ChainAdapterArgs) {
+    super(args)
+  }
+
+  getType(): ChainTypes.Cosmos {
+    return ChainTypes.Cosmos
+  }
+
+  async getAddress(input: chainAdapters.GetAddressInput): Promise<string> {
+    const { wallet, bip44Params = ChainAdapter.defaultBIP44Params } = input
+    const path = toPath(bip44Params)
+    const addressNList = bip32ToAddressNList(path)
+
+    try {
+      const cosmosAddress = await (wallet as CosmosWallet).cosmosGetAddress({
```

Comment:
would it be better to guard with `supportsCosmos` method from hdwallet rather than casting here?